city_scrapers/spiders/spo_sv_city_council.py

```diff
@@ -99,8 +100,15 @@ def _parse_links(self, item):
         for a in a_tags:
             href = a.css("::attr(href)").get()
             title = a.css("::text").get()
-            links.append({"href": href, "title": title})
+            valid_href = self._ensure_https_protocol(href)
+            links.append({"href": valid_href, "title": title})
         return links
 
+    def _ensure_https_protocol(self, url):
+        """Ensure the URL starts with 'https://', fixing incomplete
+        or incorrect parts."""
+        pattern = r"^(ttps://|tps://|ps://|s://|://|//|/)"
+        return re.sub(pattern, "https://", url, count=1)
```

Comment:
Hi @SimmonsRitchie Not sure if you should capture many patterns here. IMO I would handle 2 cases:
- If the url starts with 2 slashes`//` .eg in this case `//spokanevalley.granicus.com/AgendaViewer.php?view_id=3&event_id=1314` I will prepend it with the protocol of the current browsing site, so it's could be `http` or `https`. In this case, the `start_url` is https://spokanevalley.granicus.com/ViewPublisher.php?view_id=3 so I will extract the protocol `https` from the `start_url` and prepend it to this `url`
- If the url starts with single slash `/`, it means relative path, so I'll prepend the url with the protocol and domain name of `start_url`

I think there are some common Python libs to parse the url. Eg. in documenters project we use `urllib.parse.urlparse` to extract the protocol, domain and relative path,... I'd suggest you give it a try

Reply:
No problem, @LienDang ! I wasn't quite sure how thorough you were thinking in terms of pattern checking. I appreciate the extra detail you've provided. I've reworked the method to handle those two cases and added some tests for it. Let me know if that's okay or if you'd like additional changes 🔧

Reply:
Hi @SimmonsRitchie I don't see your changed as you said `I've reworked the method to handle those two cases and added some tests for it`. Have you updated it in another PR? It's ok to merge then.

Reply:
Hah, that was dumb of me. Sorry, @LienDang, it looks like I forgot to push my changes. Those should be there now.